src/operations/aggregate.ts

```diff
@@ -47,14 +42,16 @@ export class AggregateOperation<T = Document> extends CommandOperation<T> {
   pipeline: Document[];
   hasWriteStage: boolean;
 
-  constructor(parent: OperationParent, pipeline: Document[], options?: AggregateOptions) {
-    super(parent, options);
+  constructor(ns: MongoDBNamespace, pipeline: Document[], options?: AggregateOptions) {
+    super(undefined, { ...options, dbName: ns.db });
 
     this.options = options ?? {};
-    this.target =
-      parent.s.namespace && parent.s.namespace.collection
-        ? parent.s.namespace.collection
-        : DB_AGGREGATE_COLLECTION;
+
+    if (ns.collection === undefined || ns.collection === '') {
```

Comment:
I think we can capture the desired outcome with just `this.target = ns.collection || DB_AGGREGATE_COLLECTION` (it would preserve the previous logic, and it's different from existing logic in that it would still assign the target to `null` or `0` if those happened to get passed in as values; however, I don't see why we would want to start doing that)

Reply:
(See my comment on the other thread)